### PR TITLE
WDS Color: refine `bgAccentHover` (light mode)

### LINKED ISSUE #22823: WDS color: `bdNeutralHover` (both modes)
Perceptually not visible at all currently in light mode, barely visible in dark mode.

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Slightly darker hover for dark seeds, overall refinements in the medium, special exceptions for very light yellows. Minor fixes in utility functions.

Fixes #22823  

## Media
New on the left, current on the right
https://user-images.githubusercontent.com/80973/236816955-2be0773f-2c51-4e53-9e61-844a61ad7f2c.mov

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by somangshu — APPROVED** [review]

**Review by KelvinOm — APPROVED** [review]
LGTM. Let me know if I can merge this.

**Comment by ichik** [comment]
> LGTM. Let me know if I can merge this.

Please do.
